### ISSUE TITLE
[Enhancement]: Send email instructions when creating new aws_identitystore_user

### ISSUE THREAD

**Issue by luislhl** [hashicorp/terraform-provider-aws#28102]
### Description

![image](https://user-images.githubusercontent.com/5041650/204887121-e4c80c36-398f-43e1-899b-60dd3054fbb4.png)

**Comment by luislhl** [comment]
I just found out that there is an option that should be configured in the SSO Instance to solve this: https://docs.aws.amazon.com/console/singlesignon/directory/users-without-pwd

I'm not sure there is a way to set this through Terraform in some resource, though.

But at least there is a way even though this option lies outside the `aws_identitystore_user`.

**Comment by selfisch** [comment]
Thanks for the tip @luislhl . Thats some kind of workaround, but will not full fill the wished behavior.

When creating a user by ``aws_identitystore_user``, there is no invitation email auto send. The user has to try actively to sign in once and is then lead to an initial account config wizard.

Better then nothing, but an invitation mail on creation by the terraform resource would be nice :-).

**Comment by arvegadev** [comment]
Hello,

we are building a module that provides an easy way to manage users, groups, permission sets... and we are facing this problem. It would be nice if the terraform module could send the invitation like console provisioning does.

**Comment by arvegadev** [comment]
> I just found out that there is an option that should be configured in the SSO Instance to solve this: https://docs.aws.amazon.com/console/singlesignon/directory/users-without-pwd
> 
> I'm not sure there is a way to set this through Terraform in some resource, though.
> 
> But at least there is a way even though this option lies outside the `aws_identitystore_user`.

I tried this too, but it doesn't work. I have to send the verification email by clicking in the AWS console, which is not the ideal way to automate this kind of provisioning.

**Comment by luislhl** [comment]
It seems this is a limitation of AWS API itself: https://docs.aws.amazon.com/singlesignon/latest/IdentityStoreAPIReference/API_CreateUser.html

There is no such option on their API for Terraform to implement in the `aws_identitystore_user` resource.

**Comment by MB-MuratBayraktar** [comment]
Hello, any new updates regarding this situation?

**Comment by hynespm** [comment]
Hi, just following up on this here too, is there any update ?

**Comment by Brave-Robin** [comment]
Hello all, as I see this property is really obligatory for IAM IC. Could you add implementations of this to your plans?

**Comment by Bubjas** [comment]
Hello! Waiting for changes. Stayin online to observe this situation

**Comment by quansang** [comment]
Waiting for any update on this.. My trick now is creating all users by Terraform. After that, go to AWS Console and click `Reset password`(also verify) button for each user because only `Send email verification link` doesn't give them password

**Comment by Poupiman** [comment]
Hi everyone! Do you have some news about this feature? Some tricks to automate this process?

**Comment by be-aws-architect** [comment]
Chiming in, could really use this feature too. Built a clean automatic deploy flow.. only to have to go in the console and manually click a button..

**Comment by dsantanu** [comment]
Hi there,
doesn't look like there is any update on that yet; done anyone know any workaround to send auto-email the instruction to the users?

**Comment by dsantanu** [comment]
> I just found out that there is an option that should be configured in the SSO Instance to solve this: https://docs.aws.amazon.com/console/singlesignon/directory/users-without-pwd
> 
> I'm not sure there is a way to set this through Terraform in some resource, though.
> 
> But at least there is a way even though this option lies outside the `aws_identitystore_user`.

it definitely doesn't work when users are created by TF, even though it uses the underlaying AWS API

**Comment by Desh-Deepak-Dhobi** [comment]
It seems it's still not solved.

When the AWS Identity Center (SSO) user is created using the terraform "aws_identitystore_user", it shows "Users must first verify their email address before they can begin to use certain features such as completing email-based two-step verification during sign-in.".

![image](https://github.com/hashicorp/terraform-provider-aws/assets/59042109/d79aa109-32a7-4917-a610-c0635cbad7d0)

And to solve this I have to use the username manually and the "Forget Password" option to set up a new password and start using the created AWS SSO user.

Is there any solution found?

**Comment by dsantanu** [comment]
> And to solve this I have to use the username manually and the "Forget Password" option to set up a new password and start using the created AWS SSO user.

I think the main issue is not being able to send the `verification email`; user can reset their own password after that. I'm having huge issue with the verification email and doesn't look like it's goinhg anywhere atm.

**Comment by mattupstate** [comment]
There is a configuration option in the Identity Center instance settings that, once enabled, will result in emails being sent automatically:

<img width="1435" alt="Image" src="https://github.com/user-attachments/assets/2b950ef1-3d7e-42cd-8eb0-fbc1c605e180" />

**Comment by edmeister** [comment]
@mattupstate this is exactly the functionality that is not working.

**Comment by mattupstate** [comment]
@edmeister Yeah, I finally got around to testing it myself and learned that pretty quickly 🤦

**Comment by cciocov** [comment]
@edmeister @mattupstate The OTP code is supposed to be sent when the user first tries to login, not when you create the user from Terraform. That's what you're enabling with the above setting, which works as expected.

**Comment by taruntej-a** [comment]
Thanks for raising this. We reviewed this request and confirmed the AWS Identity Store API currently does not expose a way to trigger the user invitation email flow from CreateUser or related APIs.

Reference: https://docs.aws.amazon.com/singlesignon/latest/IdentityStoreAPIReference/API_CreateUser.html

The API lets us set email attributes on the user object, but it does not include a send invitation email option. The send email behavior appears to be console-only today. Marking this as blocked by upstream AWS API capability. If AWS adds API support for invitation email dispatch, we can revisit and implement it.
